tests/test_logging.py

```diff
@@ -0,0 +1,194 @@
+    stream = StringIO()
+    inmanta_logger = InmantaLoggerConfig.get_instance(stream)
+    handler = inmanta_logger.get_handler()
+    assert handler.level == logging.INFO
+
+    logger = logging.getLogger("test_logger")
+    expected_output = "test_logger              DEBUG   This is a test message"
+
+    # Log a message and verify that it is not logged as the log level is too high
+    logger.debug("This is a test message")
+    log_output = stream.getvalue().strip()
+    assert expected_output not in log_output
+
+    # change the log_level and verify the log is visible this time.
+    inmanta_logger.set_log_level("DEBUG")
+    logger.debug("This is a test message")
+    log_output = stream.getvalue().strip()
+    assert expected_output in log_output
+
+
+def test_set_log_formatter():
+    stream = StringIO()
+    inmanta_logger = InmantaLoggerConfig.get_instance(stream)
+    handler = inmanta_logger.get_handler()
+    assert isinstance(handler.formatter, MultiLineFormatter)
+
+    logger = logging.getLogger("test_logger")
+    expected_output_format1 = "test_logger              INFO    This is a test message"
+    expected_output_format2 = "test_logger - INFO - This is a test message"
+
+    # Log a message with the default formatter
+    logger.info("This is a test message")
+    log_output = stream.getvalue().strip()
+    assert expected_output_format1 in log_output
+
+    # change the formatter and verify the output is different
+    formatter = logging.Formatter("%(name)s - %(levelname)s - %(message)s")
+    inmanta_logger.set_log_formatter(formatter)
+    assert inmanta_logger.get_handler().formatter == formatter
+
+    logger.info("This is a test message")
+    log_output = stream.getvalue().strip()
+    assert expected_output_format2 in log_output
+
+
+def test_set_logfile_location(
+    tmpdir,
+):
+    log_file = tmpdir.join("test.log")
+    inmanta_logger = InmantaLoggerConfig.get_instance()
+    inmanta_logger.set_logfile_location(str(log_file))
+    handler = inmanta_logger.get_handler()
+    assert isinstance(handler, logging.handlers.WatchedFileHandler)
+    assert handler.baseFilename == str(log_file)
+
+    # Log a message
+    logger = logging.getLogger("test_logger")
+    logger.info("This is a test message")
+
+    # Verify the message was written to the log file
+    with open(str(log_file), "r") as f:
+        contents = f.read()
+        assert "This is a test message" in contents
+
+
+@pytest.mark.parametrize_any(
+    "log_file, log_file_level, verbose",
+    [(None, "INFO", "1"), (None, "ERROR", "4"), ("test.log", "WARNING", "4"), ("test.log", "DEBUG", "4")],
+)
+def test_apply_options(tmpdir, log_file, log_file_level, verbose):
+    stream = StringIO()
+    inmanta_logger = InmantaLoggerConfig.get_instance(stream)
+    logger = logging.getLogger("test_logger")
+
+    if log_file:
+        log_file = tmpdir.join("test.log")
+
+    options1 = Options(log_file=log_file, log_file_level=log_file_level, verbose=verbose)
```

Comment:
I think the `verbose` parameter is typed as an `int` in the `Options` class (here a string is passed)

Reply:
indeed i'll change it